gql/dsl.py

```diff
@@ -1,179 +1,364 @@
+        But with a type check for each field in the list.
+
+        :raises TypeError: if any of the provided fields are not instances
+                           of the :class:`DSLField` class.
+        """
+        ast_fields = []
+        for field in fields:
+            if isinstance(field, DSLField):
+                ast_fields.append(field.ast_field)
+            else:
+                raise TypeError(f'Received incompatible field: "{field}".')
+
+        return ast_fields
+
+    def select(self, *fields: "DSLField") -> "DSLField":
+        """Select the new children fields
+        that we want to receive in the request.
+
+        If used multiple times, we will add the new children fields
+        to the existing children fields
+
+        :param fields: new children fields
+        :type fields: DSLField
+        :return: itself
+
+        :raises TypeError: if any of the provided fields are not instances
+                           of the :class:`DSLField` class.
+        """
+
+        added_selections: List[FieldNode] = self.get_ast_fields(fields)
+
+        current_selection_set: Optional[SelectionSetNode] = self.ast_field.selection_set
+
+        if current_selection_set is None:
             self.ast_field.selection_set = SelectionSetNode(
                 selections=FrozenList(added_selections)
             )
+        else:
+            current_selection_set.selections = FrozenList(
+                current_selection_set.selections + added_selections
+            )
+
+        log.debug(f"Added fields: {fields} in {self!r}")
+
         return self
 
-    def __call__(self, **kwargs):
+    def __call__(self, **kwargs) -> "DSLField":
         return self.args(**kwargs)
 
-    def alias(self, alias):
+    def alias(self, alias: str) -> "DSLField":
+        """Set an alias
+
+        :param alias: the alias
+        :type alias: str
+        :return: itself
+        """
+
         self.ast_field.alias = NameNode(value=alias)
         return self
 
-    def args(self, **kwargs):
+    def args(self, **kwargs) -> "DSLField":
+        r"""Set the arguments of a field
+
+        The arguments are parsed to be stored in the AST of this field.
+
+        .. note::
+            you can also call the field directly with your arguments.
+            :code:`ds.Query.human(id=1000)` is equivalent to:
+            :code:`ds.Query.human.args(id=1000)`
+
+        :param \**kwargs: the arguments (keyword=value)
+        :return: itself
+
+        :raises KeyError: if any of the provided arguments does not exist
+                          for this field.
+        """
+
         added_args = []
         for name, value in kwargs.items():
             arg = self.field.args.get(name)
             if not arg:
                 raise KeyError(f"Argument {name} does not exist in {self.field}.")
-            arg_type_serializer = get_arg_serializer(arg.type, known_serializers=dict())
+            arg_type_serializer = self._get_arg_serializer(arg.type)
```

Comment:
Unrelated but I always see this `_get_arg_serializer` requires `GraphQLInputType` but this type does not include `GraphQLInputField` type.

Seems like `graphql-core` always defines it as below:

```python
GraphQLInputType = Union[
    GraphQLScalarType, GraphQLEnumType, GraphQLInputObjectType, GraphQLWrappingType
]
```

@Cito this type definition seems to be based on https://spec.graphql.org/draft/#sec-Input-Objects.Type-Validation, right?

Reply:
GraphQLInputTypes are those that may be used as input types for arguments and directives. It's based on the same type in GraphQL.js which is based on the spec you linked.

Reply:
You are right @KingDarBoja , there was a problem here because the `_get_arg_serializer` method was called with a _GraphQLField_ argument instead of a type if the parent input type was a _GraphQLInputObjectType_.
I modified the code in my last commit to get the type directly instead of using the field so that now the typing info should be correct I think.

Reply:
all this is irrelevant now that the _get_arg_serializer method was removed :rofl: